mitiq/tests/test_factories.py

```diff
@@ -123,3 +124,31 @@ def test_poly_exp_factory_no_asympt():
         algo_object = PolyExpFactory(X_VALS, order=2, asymptote=None)
         run_factory(algo_object, f)
         assert np.isclose(algo_object.reduce(), f(0, err=0), atol=CLOSE_TOL)
+
+def test_ada_exp_factory_with_asympt():
+    """Test of the adaptive exponential extrapolator."""
+    for f in [f_exp_down, f_exp_up]:
+        algo_object = AdaExpFactory(steps=3, scalar=2.0, asymptote=A)
+        run_factory(algo_object, f)
+        assert np.isclose(algo_object.reduce(), f(0, err=0), atol=CLOSE_TOL)
```

Comment:
Instead of looping over `f_exp_down` and `f_exp_up` in the body of each test, we could use [`pytest.mark.parameterize`](https://docs.pytest.org/en/latest/parametrize.html).

It won't change much here but I'm in favor of using this because:

1. It will look a little cleaner.
2. Reduce code duplication.
3. Add more unit tests :)
4. Make it easier to add more functions to the test cases in the future.

Just a suggestion, if it's a hassle for now just ignore.

Reply:
Good suggestion thanks! Now all the loops are replaced by `pytest.mark.parameterize`